backend-lib/src/main/rust/lib.rs

```diff
@@ -799,26 +799,37 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_getTotalT
     unwrap_exc_or(&mut env, res, -1)
 }
 
+fn parse_protocol(code: i32) -> Option<ShieldedProtocol> {
+    match code {
+        2 => Some(ShieldedProtocol::Sapling),
+        3 => Some(ShieldedProtocol::Orchard),
+        _ => None
+    }
+}
+
```

Comment:
Should this now be removed?

Reply:
I think that checking the protocol int on both sides of the memo API does not hurt anything and could still be useful when someone tries to use an unsupported protocol value. What do you think?

Reply:
Oh, duh, sorry I got confused. This is fine.